### ISSUE TITLE
Add AnyEditTools plugin to PoshiIDE feature

### ISSUE THREAD

**Issue by brianchiu** [gamerson/poshi-ide#22]
I looked up how to do this in eclipse, and the functionality seemed to only exist for Java files.

I installed anyedit (http://marketplace.eclipse.org/content/anyedit-tools) and now it automatically removes trailing whitespace.

Is it possible for eclipse to do this without any extra plugins?

**Comment by gamerson** [comment]
For java editor you can remove trailing whitespace automatically.  I'm not
sure if the XML editor has the same feature.  Hmm... just checked doesn't
seem so.

However, we can add anyedit-tools automatically installed in PoshiIDE

On Thu, Aug 27, 2015 at 8:21 PM, Brian Chiu notifications@github.com
wrote:

> I looked up how to do this in eclipse, and the functionality seemed to
> only exist for Java files.
> 
> I installed anyedit (http://marketplace.eclipse.org/content/anyedit-tools)
> and now it automatically removes trailing whitespace.
> 
> Is it possible for eclipse to do this without any extra plugins?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/gamerson/poshi-ide/issues/22.

## 

Greg Amerson
Liferay Developer Tools
Liferay, Inc. www.liferay.com

**Comment by gamerson** [comment]
So that you don't have to install it separately.

On Fri, Aug 28, 2015 at 10:15 AM, Greg Amerson gregory.amerson@liferay.com
wrote:

> For java editor you can remove trailing whitespace automatically.  I'm not
> sure if the XML editor has the same feature.  Hmm... just checked doesn't
> seem so.
> 
> However, we can add anyedit-tools automatically installed in PoshiIDE
> 
> On Thu, Aug 27, 2015 at 8:21 PM, Brian Chiu notifications@github.com
> wrote:
> 
> > I looked up how to do this in eclipse, and the functionality seemed to
> > only exist for Java files.
> > 
> > I installed anyedit (http://marketplace.eclipse.org/content/anyedit-tools)
> > and now it automatically removes trailing whitespace.
> > 
> > Is it possible for eclipse to do this without any extra plugins?
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > https://github.com/gamerson/poshi-ide/issues/22.
> 
> ## 
> 
> Greg Amerson
> Liferay Developer Tools
> Liferay, Inc. www.liferay.com

## 

Greg Amerson
Liferay Developer Tools
Liferay, Inc. www.liferay.com

**Comment by gamerson** [comment]
This will require two steps
1. add AnyEdit repository to parent pom
2. add an "included feature" to the Poshi IDE feature.xml, like this:

```
<requires>
      <import feature="<any-edit-feature-id>" version="<latest-version>" match="greaterOrEqual"/>
</requires>
```
1. go to the poshi ide repository and edit the hidden features.product.  And add a new entry under the existing feature to add a new line for AnyEdit feature.

**Comment by jtydhr88** [comment]
Hey Greg,
It doesn't work now, if you have time, please help me to fix it, thanks.
you can build by "mvn clean verify" to see problem

**Comment by gamerson** [comment]
I see the problem.  AnyEdit updatesite is the old style updatesite with
site.xml which doesn't work with tycho.  We are going to need to convert
this to a p2 style repository.  Please see this for how to perform this
task:

https://wiki.eclipse.org/Tycho/Additional_Tools#publish-features-and-bundles_goal

On Mon, Sep 21, 2015 at 7:26 PM, Terry Jia notifications@github.com wrote:

> Hey Greg,
> It doesn't work now, if you have time, please help me to fix it, thanks.
> you can build by "mvn clean verify" to see problem
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/gamerson/poshi-ide/issues/22#issuecomment-142146143.

## 

Greg Amerson
Liferay Developer Tools
Liferay, Inc. www.liferay.com

**Comment by gamerson** [comment]
Merged, thanks! d8e35c2...f990c50 <br><br>:octocat: _Sent from [GH](http://nodegh.io)._
